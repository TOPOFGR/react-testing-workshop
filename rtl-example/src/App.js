import logo from './logo.svg';
import './App.css';
import Mocked from './components/Mocked';

function App() {

  const onSubmit = () => {
    console.log('onSubmit');
  }

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
        <Mocked onSubmit={onSubmit}/>
      </header>
    </div>
  );
}

export default App;
