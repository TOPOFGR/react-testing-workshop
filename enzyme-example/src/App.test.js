import { render, mount } from 'enzyme';

import App from "./App";

// jest.mock("./components/Mocked");

describe("App", () => {
  test("renders without crashing", () => {
    const view = render(<App />);

    expect(view).toMatchSnapshot();
  });

  test("renders learn react link", () => {
    const wrapper = mount(<App />);

    const linkElement = wrapper.find('a');

    expect(linkElement).toHaveLength(1);
  });
});
