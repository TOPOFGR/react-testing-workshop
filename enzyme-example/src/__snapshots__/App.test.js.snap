// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
LoadedCheerio {
  "0": <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.js
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
      <div>
        Mocked component
        <form>
          <input
            type="text"
          />
          <button>
            Save
          </button>
        </form>
      </div>
    </header>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
